--- a/src/meson.build
+++ b/src/meson.build
@@ -66,8 +66,7 @@
 		variables: [
 			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
 			'pkgdatadir=' + join_paths('${pc_sysrootdir}${datarootdir}', meson.project_name()),
-			'bindir=' + join_paths('${prefix}', get_option('bindir')),
-			'wayland_scanner=${bindir}/wayland-scanner'
+			'wayland_scanner=' + join_paths('${prefix}', get_option('bindir'), 'wayland-scanner')
 		],
 		filebase: 'wayland-scanner'
 	)
